spec/features/visit_a_vacancy_page_spec.rb

```diff
@@ -8,8 +8,6 @@
 
     visit root_path
 
-    click_link I18n.t('messages.apply_for_job')
```

Comment:
This is wrong. The test now doesn't do anything.

I think you should replace this by:

```
click_link vacancy.job_title
```

So the test will visit the right page instead of keeping it in the root page

Reply:
or something like this. I don't know how Capybara will behave since the entire div is clickable. Maybe you will need to add an identifier on the div and use it here to identify. I can try help you if you want